src/main/kotlin/gg/flyte/twilight/extension/Entity.kt

```diff
@@ -27,4 +29,196 @@ fun Entity.getNearbyEntities(range: Double): MutableList<Entity> {
+            nearestPlayer = player
+        }
+    }
+    return nearestPlayer
+}
+
+
+/**
+ * Gets the nearest player to this entity within the given radius.
+ * @param radius The radius to search.
+ * @return The nearest player.
+ */
+
+fun Entity.getNearestPlayer(radius: Double): Player? {
+    var nearestDistance = Double.MAX_VALUE
+    var nearestPlayer: Player? = null
+    for (player in world.getNearbyPlayers(location, radius)) {
+        if (player == this) continue
+        val distance = location.distance(player.location)
+        if (distance < nearestDistance) {
+            nearestDistance = distance
+            nearestPlayer = player
+        }
+    }
+    return nearestPlayer
+}
+
+
+/**
+ * Gets the nearest player to this entity within the given radius.
+ * @param xRadius The x radius to search.
+ * @param yRadius The y radius to search.
+ * @param zRadius The z radius to search.
+ * @return The nearest player.
+ */
+
+fun Entity.getNearestPlayer(xRadius: Double, yRadius: Double, zRadius: Double): Player? {
+    var nearestDistance = Double.MAX_VALUE
+    var nearestPlayer: Player? = null
+    for (player in world.getNearbyPlayers(location, xRadius, yRadius, zRadius)) {
+        if (player == this) continue
+        val distance = location.distance(player.location)
+        if (distance < nearestDistance) {
+            nearestDistance = distance
+            nearestPlayer = player
+        }
+    }
+    return nearestPlayer
+}
+
+
+/**
+ * Gets the nearest entity to this entity.
+ * @return The nearest entity.
+ */
+
+fun Entity.getNearestEntity(): Entity? {
+    var nearestDistance = Double.MAX_VALUE
+    var nearestEntity: Entity? = null
+    for (entity in world.entities.also { it.remove(this) }) {
+        if (entity == this) continue
+        val distance = location.distance(entity.location)
+        if (distance < nearestDistance) {
+            nearestDistance = distance
+            nearestEntity = entity
+        }
+    }
+    return nearestEntity
+}
+
+
+/**
+ * Gets the nearest entity to this entity within the given radius.
+ * @param radius The radius to search.
+ * @return The nearest entity.
+ */
+
+fun Entity.getNearestEntity(radius: Double): Entity? {
+    var nearestDistance = Double.MAX_VALUE
+    var nearestEntity: Entity? = null
+    for (entity in world.getNearbyEntities(location, radius, radius, radius)) {
+        if (entity == this) continue
+        val distance = location.distance(entity.location)
+        if (distance < nearestDistance) {
+            nearestDistance = distance
+            nearestEntity = entity
+        }
+    }
+    return nearestEntity
+}
+
+
+/**
+ * Gets the nearest entity to this entity within the given radius.
+ * @param xRadius The x radius to search.
+ * @param yRadius The y radius to search.
+ * @param zRadius The z radius to search.
+ * @return The nearest entity.
+ */
+
+fun Entity.getNearestEntity(xRadius: Double, yRadius: Double, zRadius: Double): Entity? {
```

Comment:
You can set default values here so you dont need multiple methods that do the same thing